### PR TITLE
1704: Update the Practitioner Details to get the Groups and Practitioner Role resources attached to a practitioner.

### PR DESCRIPTION
**Pull request by rehammuzzamil**
https://github.com/opensrp/fhircore/issues/1704

### CODE REVIEW

File: practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java [hunk 1]
```diff
@@ -450,6 +466,20 @@ private List<String> getLocationIdentifiersByIds(List<Long> locationIds) {
         return locationsIdentifiers;
     }
 
+    private List<IBaseResource> getP(Long practitionerId) {
```
Comment (dubdabasoduba) [inline practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java]: Please name the variables better. Variables show be self-documenting e.g `getParticipant`

**Review by dubdabasoduba — DISMISSED** [review]

File: practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java [hunk 2]
```diff
@@ -346,6 +356,26 @@ private List<Organization> mapToTeams(List<IBaseResource> teams) {
...
 
+    private List<PractitionerRole> mapToPractitionerRoles(List<IBaseResource> practitionerRoles) {
+        List<PractitionerRole> practitionerRoleList = new ArrayList<>();
+        PractitionerRole practitionerRoleObj;
+        for (IBaseResource practitionerRole : practitionerRoleList) {
+            practitionerRoleObj = (PractitionerRole) practitionerRole;
+            practitionerRoleList.add(practitionerRoleObj);
+        }
+        return practitionerRoleList;
+    }
```
Comment (dubdabasoduba) [inline practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java]: @rehammuzzamil the `practitionerRoles` param doesn't seem to be used. This means the `practitionerRoleList` will always a size of 0.

File: practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java [hunk 3]
```diff
@@ -450,6 +480,22 @@ private List<String> getLocationIdentifiersByIds(List<Long> locationIds) {
...
+        code.setValue("405623001");
+        code.setSystem("http://snomed.info/sct");
+        coding.add(code);
+        codeListParam.addAnd(coding);
+        groupSearchParameterMap.add(CODE, codeListParam);
+        IBundleProvider groupsBundle = groupIFhirResourceDao.search(groupSearchParameterMap);
+        return groupsBundle != null
+                ? groupsBundle.getResources(0, groupsBundle.size())
+                : new ArrayList<>();
+    }
```
Comment (dubdabasoduba) [inline practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java]: @rehammuzzamil The `practitionerId` is also not used here. The code below does not account for the Practitioner Id on the search parameters.

**Comment by dubdabasoduba** [comment]
@rehammuzzamil The `mvn clean install` fails because of some tests. Let's try to fix this.

**Review by dubdabasoduba — CHANGES_REQUESTED** [review]

File: practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java (hunk 2 above)
Comment (rehammuzzamil) [reply practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java]: Done

**Comment by rehammuzzamil** [comment]
> @rehammuzzamil The `mvn clean install` fails because of some tests. Let's try to fix this.

Switched to Mockito core, as Mockito-all is deprecated @dubdabasoduba

File: practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java (hunk 3 above)
Comment (rehammuzzamil) [reply practitioner/src/main/java/org/smartregister/extension/rest/PractitionerDetailsResourceProvider.java]: Updated

**Review by dubdabasoduba — APPROVED** [review]

**Comment by rehammuzzamil** [comment]
The PR also holds the fix for https://github.com/opensrp/fhircore/issues/1995

**Pull request merged**